src/main/java/com/mykaarma/urlshortener/service/UrlService.java

```diff
@@ -114,15 +114,44 @@ public UrlDetails shortenUrl(String longUrl, String shortUrlDomain, long expiryD
 		log.info(String.format("Creating a new shortUrl for longUrl=%s and businessUUID=%s", longUrl, businessUUID));
 		int retryCount = 1;
 		UrlDetails shortUrlDetails;
-		shortUrlDetails = createShortUrl(shortUrlDomain, longUrl, expiryDate, businessUUID, additionalParams, urlPrefix, retryCount);
+		shortUrlDetails = createShortUrl(shortUrlDomain, longUrl, expiryDate, businessUUID, additionalParams, urlPrefix, retryCount,overwrite);
 
 		return shortUrlDetails;
 
 	}
 
-	private UrlDetails createShortUrl(String shortUrlDomain, String longUrl, Date expiryDate, String businessUUID, Map<String, String> additionalParams,String urlPrefix,int retryCount) {
+	private UrlDetails createShortUrl(String shortUrlDomain, String longUrl, Date expiryDate, String businessUUID, Map<String, String> additionalParams,String urlPrefix,int retryCount,boolean overwrite) {
 
 		log.info(String.format("Creating a new shortUrl using retry method  for longUrl=%s and businessUUID=%s", longUrl, businessUUID));
+		UrlDetails existingShortUrl = urlRepository.getActiveUrlDetailsByLongUrlAndBusinessUUIDAndDomain(longUrl, businessUUID, shortUrlDomain);
+
+		if (existingShortUrl != null) {
+
+			log.info(String.format("ShortUrl=%s already present for longUrl=%s businessUUID=%s shortUrlDomain=%s", existingShortUrl.getShortUrl(), existingShortUrl.getLongUrl(), existingShortUrl.getBusinessUUID(), existingShortUrl.getShortUrlDomain()));
+
+			if (overwrite) {
+				if (expiryDate.before(existingShortUrl.getExpiryDateTime())) {
+					log.error(String.format("The provided expiryDate for shortUrl=%s is before the existing expiryDate", existingShortUrl.getShortUrl()));
```

Comment:
What should happen in this case?

Reply:
we are not updating the expiry date to the database if it is before the expiryDate, the epxiryDate in the DB would be the one which has longer duration